logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java

```diff
@@ -70,24 +76,28 @@ public static MQProducer getInstance(String nameServerAddress, String group) thr
     }
 
 
-    public static void removeAndClose(String nameServerAddress, String group) {
+    public void removeAndClose(String nameServerAddress, String group) {
         if (group == null) {
             group = DEFAULT_GROUP;
         }
         String genKey = genKey(nameServerAddress, group);
-        MQProducer producer = producerMap.remove(genKey);
+        MQProducer producer = getProducerInstance().producerMap.remove(genKey);
 
         if (producer != null) {
             producer.shutdown();
         }
     }
 
-    public static void closeAll() {
-        Set<Map.Entry<String, MQProducer>> entries = producerMap.entrySet();
+    public void closeAll() {
+        Set<Map.Entry<String, MQProducer>> entries = getProducerInstance().producerMap.entrySet();
         for (Map.Entry<String, MQProducer> entry : entries) {
-            producerMap.remove(entry.getKey());
+            getProducerInstance().producerMap.remove(entry.getKey());
             entry.getValue().shutdown();
         }
     }
 
+    public static void setInstance(ProducerInstance instance1) {
```

Comment:
looks pretty dangerous to me.
Especially considering you make it only for tests.